rocksdb-sys/build.rs

```diff
@@ -18,11 +18,22 @@ fn main() {
 
         println!("cargo:rustc-link-lib=dylib={}", "rpcrt4");
         println!("cargo:rustc-link-lib=dylib={}", "shlwapi");
+
+        let features = env::var("CARGO_CFG_TARGET_FEATURE").unwrap_or_default();
+        if features.contains("crt-static") {
+            cfg.define("WITH_MD_LIBRARY", "OFF");
+        }
     } else {
         cfg.define("SNAPPY_INCLUDE_DIR", snappy)
             .define("SNAPPY_LIBRARIES", "/dev/null");
     }
 
+    // NOTE: the cfg! macro doesn't work when cross-compiling, it would return values for the host
+    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap_or_default();
+    if target_arch == "arm" || target_arch == "aarch64" {
```

Comment:
IIUC, we assume that the host arch is `x86_64`? And, ideally, we should check for `host_arch != target_arch`? But I'm not sure if there is a simple way to find out the value of the `host_arch`.

Reply:
Indeed, I think `-march=native` would work if the host was ARM and it was building for ARM, but the crate should build on ARM nonetheless with the current changes.
The current build file isn't terribly portable and yes it makes some assumptions regarding the build environment, i.e. my priority was to make it work on parity-ethereum's CI.